### PR TITLE
Fix PDF generation for custom reports, Fix yarn dependancies, Add shared media volume for downloadable reports, Add some error handeling in models, Revisions to reporting widgets and forms

### PR DESCRIPTION
**Pull request by sscottgvit**
Please submit your pull requests to the 'dev' branch.

When submitting a pull request, please make sure you have completed the following checklist:

- [x] Your code is flake8 compliant (DefectDojo's code isn't currently flake8 compliant, but we're trying to correct that.)
- [x] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.
- [x] Model changes should include the necessary migrations in the dojo/dd_migrations folder.
- [x] Add applicable tests to the unit tests.

### CODE REVIEW

**Comment by sscottgvit** [comment]
snyk job won't pass for some odd reason. Am I missing something?

"snyk monitor requires an authenticated account. Please run `snyk auth` and try again."

**Comment by valentijnscholten** [comment]
> snyk job won't pass for some odd reason. Am I missing something?
> 
> "snyk monitor requires an authenticated account. Please run `snyk auth` and try again."

As far as I know it is doing this for all PRs for some time now and is ignored for the "All checks have passed" thing.

File: dojo/tasks.py [hunk 1]
```diff
@@ -184,26 +184,27 @@ def async_custom_pdf_report(self,
             x = urlencode({'title': cp.title,
                            'subtitle': cp.sub_heading,
                            'info': cp.meta_info})
-            cover = host + reverse(
+            cover = "http://nginx:8080/" + reverse(
```
Comment (valentijnscholten) [inline dojo/tasks.py]: Are we sure this hardcoded url is what we want here?

Comment (aaronweaver) [reply dojo/tasks.py]: Agree, it should be pulled from settings.py and with a default value.

File: dojo/templates/dojo/custom_pdf_report.html [hunk 2]
```diff
@@ -21,7 +21,7 @@
     <meta name="pdfkit-debug-javascript" content=""/>
     <meta name="pdfkit-quiet" content=""/>
     <title>{{ report_name }}</title>
-    <link href="{{ host }}{% static "bootswatch-dist/css/bootstrap.css" %}" rel="stylesheet">
+    <link href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css" rel="stylesheet">
```
Comment (aaronweaver) [inline dojo/templates/dojo/custom_pdf_report.html]: Can we pull this in via yarn, we try to keep all dependencies local.

File: dojo/tasks.py (hunk 1 above)
Comment (sscottgvit) [reply dojo/tasks.py]: Sure, I can do that. What nginx:8080 does in this case is allow for the celery worker to reach the nginx container by resolving its IP automatically. What would you like the parameter to be listed as under settings?

File: dojo/templates/dojo/custom_pdf_report.html (hunk 2 above)
Comment (sscottgvit) [reply dojo/templates/dojo/custom_pdf_report.html]: Sure, I'll tweak it.

File: dojo/tasks.py (hunk 1 above)
Comment (sscottgvit) [reply dojo/tasks.py]: Moved into settings. Also added param in settings to switch between containerized and non-containerized path. At the moment this only matters for things that run in the celery_worker that need to hit nginx.

File: dojo/templates/dojo/custom_pdf_report.html (hunk 2 above)
Comment (sscottgvit) [reply dojo/templates/dojo/custom_pdf_report.html]: Resolved.

**Comment by sscottgvit** [comment]
Let me know if there's anything else I need to do for this PR. Thanks all!

**Comment by Maffooch** [comment]
Theres been a ton of changes very recently! Please fix merge conflicts.

**Pull request closed without merging**